### ISSUE TITLE
Flight Icon Request: loading and run

### ISSUE THREAD

**Issue by amyrlam** [hashicorp/flight#282]
**Outline the use case for the new icon.**
1:1 replacement for existing structure icons

**If there's a similar existing icon, why isn't it a good fit? Please describe.**
Flight doesn't have animated icons rn

**Describe alternatives you've considered.**
N/A

**Additional context**
- Command - F on this page to see the legacy structure icon `loading` that's used in `cloud-ui` production currently: https://unpkg.fly.dev/@hashicorp/structure-icons@1.9.2/dist/index.html
- `run` is the only other icon without a mapping. Am not sure where it's being used yet
- Mapping structure to flight: https://github.com/hashicorp/flight/blob/main/structure-mappings.json#L79
- Shared by @meirish on Slack

**Comment by meirish** [comment]
I _think_ the run icon is used on TFC run states

**Comment by jesdavpet** [comment]
Just wanted to `+1` @amyrlam  on the need for parity with an animated `loading` icon, WRT this discussion on Slack:
https://hashicorp.slack.com/archives/CPB8GS9QT/p1638825012141200?thread_ts=1638823666.137000&cid=CPB8GS9QT

TL;DR
- lack of a `loading` icon necessitates mixing both old and new icon libraries
- mixing icon libraries results in inconsistent and visually jarring spacing differences
- working around this with hacky one-off CSS is not ideal

**Comment by amyrlam** [comment]
Context:
- It sounds like the `loading` icon is used more, so it is likely higher priority over `run`.
- Note that in Structure, `loading` is its own component: https://structure.hashicorp.vercel.app/?path=/story/components-loading--index
- But some of the styling for `loading` is in structure-icons https://github.com/hashicorp/structure-icons/blob/91a2d6116b716ef109becec73502263066a9062d/README.md#updating-icons

Next steps (I think these are the next steps):
- @cveigt will start working on this issue but a due date is unclear
- For now it is preferable to use a manual CSS override for Structure `loading`, in conjunction with Flight icons.

**Comment by didoo** [comment]
Some things I noticed and I posted in an internal thread, and I am reposting here for context (it is not exhaustive, please take it just as extra info):

in [Structure library in Figma](https://www.figma.com/file/B7yd8loYS7nA2TiQcJ8UtV/Structure-1.9?node-id=3576%3A6057) seems there are two icons, `loading` and `run`

<img width="591" alt="screenshot_489" src="https://user-images.githubusercontent.com/686239/145094100-225579f4-6887-4faf-abb9-f173d68b89af.png">

how the animation for the `loading` icon is done in Structure:
https://github.com/hashicorp/structure/blob/main/packages/pds-ember/public/structure-icons/loading.css
there are two animations (`fancy` and `simple`) controlled via a media-query

this code is duplicated here:
https://github.com/hashicorp/structure-icons/blob/a3d53fbca427c54abcca216974834648ded89a5f/src/loading.css
not sure why

this is how the animation is done for the `run` icon
https://github.com/hashicorp/structure-icons/blob/a3d53fbca427c54abcca216974834648ded89a5f/src/run.css

**Comment by sabrinako** [comment]
+1 for `loading`, the Waypoint UI uses this Structure icon. Just wanted to add another data point 🙂

**Comment by cveigt** [comment]
**Update:** We are bringing both "loading" and "running" icons to the next version of Flight.

<img width="540" alt="Screen Shot 2022-01-14 at 1 16 09 PM" src="https://user-images.githubusercontent.com/2142312/149586302-9a935e8a-a49a-47fe-ab66-1b9d27470cf9.png">

**For more context**

We (the DS team) talked to some product designers and have taken into consideration a variety of [use cases](https://www.figma.com/file/7gaYhKwv7RKD6c5QosycPa/%E2%98%A0%EF%B8%8FGraveyard?node-id=23%3A13699) through HCP and TFC, as well as [their feedback](https://hashicorp.slack.com/archives/C025N5V4PFZ/p1641843726014200) during the design phase.

They can be found already within Figma in the [latest release](https://hashicorp.slack.com/archives/C7KTUHNUS/p1642189882014100) the team made today. 

<img width="265" alt="Screen Shot 2022-01-14 at 1 21 40 PM" src="https://user-images.githubusercontent.com/2142312/149586873-7c18aebe-4697-429c-a317-382dc58fb272.png">

We'll be adding some documentation to the [website](https://flight-hashicorp.vercel.app/) regarding usage soon.

**Side notes**
@didoo here's the [Codepen](https://codepen.io/cveigt/pen/xxXJmML) for reference on the animation.

On Accessibility: On the eng side, we should add a media query to support `prefers-reduced-motion`, though I'd like @MelSumner to confirm this. Don't think the icons need alternate styling, `animation: none` would be enough.
